core/src/metrics.rs

```diff
@@ -0,0 +1,162 @@
+//! Metrics and status reporting
+
+use std::{sync::Arc, time::SystemTime};
+
+use eyre::{Result, WrapErr as _};
+use iroha_telemetry::metrics::Metrics;
+use parking_lot::Mutex;
+use storage::storage::StorageReadOnly;
+
+use crate::{
+    kura::Kura,
+    queue::Queue,
+    state::{State, StateReadOnly, WorldReadOnly},
+    IrohaNetwork,
+};
+
+/*
+The values in the following struct are not atomics because the code that
+operates on them assumes their values does not change during the course of
+the function.
+*/
+#[derive(Debug)]
+struct LastUpdateMetricsData {
+    block_height: u64,
+}
+
+/// Responsible for collecting and updating metrics
+#[derive(Clone)]
+pub struct MetricsReporter {
+    state: Arc<State>,
+    network: IrohaNetwork,
+    kura: Arc<Kura>,
+    queue: Arc<Queue>,
+    metrics: Metrics,
+    last_update_metrics_mutex: Arc<Mutex<LastUpdateMetricsData>>,
+}
+
+impl MetricsReporter {
+    /// Construct [`Self`]
+    pub fn new(
+        state: Arc<State>,
+        network: IrohaNetwork,
+        kura: Arc<Kura>,
+        queue: Arc<Queue>,
+    ) -> Self {
+        Self {
+            state,
+            network,
+            queue,
+            kura,
+            metrics: Metrics::default(),
+            last_update_metrics_mutex: Arc::new(Mutex::new(LastUpdateMetricsData {
+                block_height: 0,
+            })),
```

Comment:
```suggestion
            last_update_metrics_mutex: <_>::default(),
```

You can impl `Default` for `LastUpdateMetricsData` and replace a whole thing with `<_>::default()`.